core/templates/dev/head/domain/collection/CollectionObjectFactory.js

```diff
@@ -192,6 +211,7 @@ oppia.factory('CollectionObjectFactory', [
     // skills from all collections nodes within this domain object. Please note
     // this operation returns a new skill list everytime this function is
     // called.
+    // TODO(wxy): migrate this to use the skills dict instead.
```

Comment:
Is the ``SkillListObjectFactory`` still being used then? If not, we don't want to introduce dead code in this migration, I think. If it simplifies the editor, then perhaps the list object should also be migrated.